metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -30,19 +30,52 @@ func (r *KeptnPrometheusProvider) EvaluateQuery(ctx context.Context, metric metr
 		return "", nil, err
 	}
 	api := prometheus.NewAPI(client)
-	result, w, err := api.Query(
-		ctx,
-		metric.Spec.Query,
-		queryTime,
-		[]prometheus.Option{}...,
-	)
+	var result model.Value
+	var warnings prometheus.Warnings
+	
+	if metric.Spec.Range != nil {
```

Comment:
I suggest to handle the different cases (with time range, without time range) in separate functions to keep this method easy to read. The log message indicating which query is being executed should then also moved into those (currently the generated query is logged a couple of lines above, but this will change depending if there is a time range or not)